pkg/operator/controllers/pullsecret/pullsecret_controller.go

```diff
@@ -78,13 +84,53 @@ func (r *PullSecretReconciler) Reconcile(request ctrl.Request) (ctrl.Result, err
 			ps.Data = map[string][]byte{}
 		}
 
-		// validate
-		if !json.Valid(ps.Data[v1.DockerConfigJsonKey]) {
+		keyCondition := &status.Condition{
+			Type:   aro.RedHatKeyPresent,
+			Status: v1.ConditionFalse,
+			Reason: "CheckFailed",
+		}
+		samplesCondition := &status.Condition{
+			Type:   aro.SamplesOperatorModified,
+			Status: v1.ConditionFalse,
+			Reason: "RedHatKeyNotPresent",
+		}
+
+		// parse keys and validate JSON
+		parsedKeys, err := r.parseRHRegistryKeys(ps)
+		foundKeys := []string{}
+		if err != nil {
 			r.log.Info("pull secret is not valid json - recreating")
 			delete(ps.Data, v1.DockerConfigJsonKey)
+			keyCondition.Reason = "CheckDone"
+			keyCondition.Message = "Cannot parse secret data, recreating."
+		} else {
+			foundKeys = r.checkRHRegistryKeys(parsedKeys)
+			keyCondition = r.updateRHKeysCondition(foundKeys)
+		}
+
+		updated := false
+		if len(foundKeys) > 0 {
+			// enable samples operator
+			updated, err = r.switchSamples(ctx, true)
+			samplesCondition.Reason = "RedHatKeyPresent"
+		} else {
+			// disable samples operator
+			updated, err = r.switchSamples(ctx, false)
```

Comment:
We tripped with Prometheus on this. I would like to see feature gate for this in ARO CRD `Features FeaturesSpec `json:"features,omitempty"`` so we can control behaviour for this. And potentially set to false for the first release, and set to true in the next one. 
```
// ClusterSpec defines the desired state of Cluster
type ClusterSpec struct {
	// ResourceID is the Azure resourceId of the cluster
	ResourceID      string              `json:"resourceId,omitempty"`
	ACRName         string              `json:"acrName,omitempty"`
	ACRDomain       string              `json:"acrDomain,omitempty"`
	Location        string              `json:"location,omitempty"`
	GenevaLogging   GenevaLoggingSpec   `json:"genevaLogging,omitempty"`
	InternetChecker InternetCheckerSpec `json:"internetChecker,omitempty"`

	Features FeaturesSpec `json:"features,omitempty"`
}

// FeaturesSpec defines ARO operator features gates flags
type FeaturesSpec struct {
	ManageSamplesOperator bool `json:"location,omitempty"`
}
```

Having this will enable to have E2E tests too for this, not only unit tests. But if enough unit tests with fake k8s and checking can be done, happy to ignore lack of E2E.